test/governance/Governor.test.js

```diff
@@ -663,349 +621,375 @@ contract('Governor', function (accounts) {
+              .withArgs(
+                this.proposal.id,
+                Enums.ProposalState.Executed,
+                GovernorHelper.proposalStatesToBitMap([Enums.ProposalState.Pending]),
+              );
           });
         });
       });
 
       describe('proposal length', function () {
         it('empty', async function () {
           this.helper.setProposal([], '<proposal description>');
-          await expectRevertCustomError(this.helper.propose(), 'GovernorInvalidProposalLength', [0, 0, 0]);
+
+          await expect(this.helper.propose())
+            .to.be.revertedWithCustomError(this.mock, 'GovernorInvalidProposalLength')
+            .withArgs(0, 0, 0);
         });
 
         it('mismatch #1', async function () {
           this.helper.setProposal(
             {
               targets: [],
-              values: [web3.utils.toWei('0')],
-              data: [this.receiver.contract.methods.mockFunction().encodeABI()],
+              values: [0n],
+              data: [this.receiver.interface.encodeFunctionData('mockFunction')],
             },
             '<proposal description>',
           );
-          await expectRevertCustomError(this.helper.propose(), 'GovernorInvalidProposalLength', [0, 1, 1]);
+          await expect(this.helper.propose())
+            .to.be.revertedWithCustomError(this.mock, 'GovernorInvalidProposalLength')
+            .withArgs(0, 1, 1);
         });
 
         it('mismatch #2', async function () {
           this.helper.setProposal(
             {
-              targets: [this.receiver.address],
+              targets: [this.receiver.target],
               values: [],
-              data: [this.receiver.contract.methods.mockFunction().encodeABI()],
+              data: [this.receiver.interface.encodeFunctionData('mockFunction')],
             },
             '<proposal description>',
           );
-          await expectRevertCustomError(this.helper.propose(), 'GovernorInvalidProposalLength', [1, 1, 0]);
+          await expect(this.helper.propose())
+            .to.be.revertedWithCustomError(this.mock, 'GovernorInvalidProposalLength')
+            .withArgs(1, 1, 0);
         });
 
         it('mismatch #3', async function () {
           this.helper.setProposal(
             {
-              targets: [this.receiver.address],
-              values: [web3.utils.toWei('0')],
+              targets: [this.receiver.target],
+              values: [0n],
               data: [],
             },
             '<proposal description>',
           );
-          await expectRevertCustomError(this.helper.propose(), 'GovernorInvalidProposalLength', [1, 0, 1]);
+          await expect(this.helper.propose())
+            .to.be.revertedWithCustomError(this.mock, 'GovernorInvalidProposalLength')
+            .withArgs(1, 0, 1);
         });
       });
 
       describe('frontrun protection using description suffix', function () {
+        const shouldPropose = () => {
```

Comment:
I'd keep this one as a regular function instead of an arrow function. The reason is that Mocha itself [discourages passing arrow functions](https://github.com/mochajs/mocha/blob/eca4fec9eee1332a0474b80aa1740822438955f3/docs/index.md#arrow-functions).

Here we don't need the `this` context outside of the `it` blocks, but I think we should follow for consistency.

```suggestion
        function shouldPropose() {
```